pricelist_item_generator/models/pricelist.py

```diff
@@ -0,0 +1,415 @@
+        string='Based on', required=True, size=-1,
+        help="Base price for computation.")
+    base_pricelist_id = fields.Many2one(
+        comodel_name='product.pricelist', string='Other PL',
+        help="Other pricelist on which this item is based.")
+    price_surcharge = fields.Float(
+        string='Price Surcharge',
+        digits_compute=dp.get_precision('Product Price'),
+        help='Specify the fixed amount to add or substract(if negative) to '
+             'the amount calculated with the discount.')
+    price_discount = fields.Float(
+        string='Discount (%)', digits=(16, 2), default=0)
+    price_round = fields.Float(
+        string='Price Rounding',
+        digits_compute=dp.get_precision('Product Price'),
+        help="Sets the price so that it is a multiple of this value.\n"
+             "Rounding is applied after the discount and before the "
+             "surcharge.\n To have prices that end in 9.99, "
+             "set rounding 10, surcharge -0.01")
+    price_min_margin = fields.Float(
+        string='Min. Price Margin',
+        digits_compute=dp.get_precision('Product Price'),
+        help='Specify the minimum amount of margin over the base price.')
+    price_max_margin = fields.Float(
+        string='Max. Price Margin',
+        digits_compute=dp.get_precision('Product Price'),
+        help='Specify the maximum amount of margin over the base price.')
+    todo = fields.Selection(
+        selection=TODO_SELECTION, readonly=True, string='Next Action',
+        help="Required action on Pricelist Items")
+
+    @api.one
+    @api.constrains('price_min_margin', 'price_max_margin')
+    def _check_margin(self):
+        if self.price_max_margin and self.price_min_margin and (
+                self.price_min_margin > self.price_max_margin):
+            raise UserError(_("Error! The minimum margin should be lower "
+                              "than the maximum margin."))
+
+    @api.one
+    @api.constrains('price_discount', 'price_surcharge')
+    def _check_price_elements(self):
+        # check price validity
+        if self.price_discount == 0 and self.price_surcharge == 0:
+            raise UserError(
+                _("'Discount' or 'Surcharge' must be different of 0."))
+
+    @api.model
+    def _price_field_get(self):
+        # TODO future version : support other type of pricelists
+        result = []
+        result.extend([(1, _('Public Price')), (-1, _('Other Pricelist'))])
+        return result
+
+
+class PricelistItemTemplate(models.Model):
+    """ This model is called by a second class
+    """
+    _inherit = ['abstract.pricelist.item.generator', 'pricelist.item.template']
+    _name = 'pricelist.item.template'
+
+    _order = 'sequence ASC, id ASC'
+
+    price_generator_id = fields.Many2one(
+        comodel_name='pricelist.item.generator', required=True)
+
+    @api.model
+    def create(self, vals):
+        self.env['pricelist.item.generator'].search(
```

Comment:
You can use browse here instead of search - `...].browse(vals['price_generator_id'])...`